src/Orleans/Utils/Pool/PooledResource.cs

```diff
@@ -54,7 +32,7 @@ public virtual void SignalPurge()
         /// </summary>
         public void Dispose()
         {
-            IObjectPool<T> localPool = Interlocked.Exchange(ref pool, null);
```

Comment:
I think that the type doesn't matter here because variable name contains all needed information for the user.